contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1180,6 +1189,90 @@ tle_name_comparison(const char *tlename, const char *identifier)
 		return (0 == strcmp(tlename, identifier));
 }
 
+/*
+* A special case of the get_object_address_relobject() function, specifically
+* for the case of triggers in tsql dialect. We add a pg_trigger lookup to search
+* for the relation that the trigger is associated with, since the relation name
+* is not supplied by the user, and thus not a part of the *object list.
+*/
+static ObjectAddress
+get_trigger_object_address(List *object, Relation *relp, bool missing_ok)
+{
+	ObjectAddress address;
+	Relation	relation = NULL;
+	const char *depname;
+	Oid			reloid;
+	Relation	tgrel;
+	ScanKeyData		key;
+	SysScanDesc tgscan;
+	HeapTuple	tuple;
+
+	/* Extract name of dependent object. */
+	depname = strVal(llast(object));
+
+	if (prev_get_trigger_object_address_hook)
+		return (*prev_get_trigger_object_address_hook)(object,relp,missing_ok);
+
+	/* 
+	* Get the table name of the trigger from pg_trigger. We know that
+	* trigger names are forced to be unique in the tsql dialect, so we
+	* can rely on searching for trigger name to find the corresponding
+	* relation name.
+	*/
+	tgrel = table_open(TriggerRelationId, AccessShareLock);
+
+	ScanKeyInit(&key,
+					Anum_pg_trigger_tgname,
+					BTEqualStrategyNumber, F_NAMEEQ,
+					CStringGetDatum(depname));
+
+	tgscan = systable_beginscan(tgrel, TriggerRelidNameIndexId, false,
+									NULL, 1, &key);
+
+	while (HeapTupleIsValid(tuple = systable_getnext(tgscan)))
+	{
+		Form_pg_trigger pg_trigger = (Form_pg_trigger) GETSTRUCT(tuple);
+		char *pg_trigger_schema_name = get_namespace_name(get_rel_namespace(pg_trigger->tgrelid));
+		char *cur_schema_name = get_dbo_schema_name(get_cur_db_name());
+		if (namestrcmp(&(pg_trigger->tgname), depname) == 0 && 
+			namestrcmp(pg_trigger_schema_name,cur_schema_name) == 0)
```

Comment:
why do we assume that current schema is DBO only?

Reply:
For other schema, if delete it , then the sql it'll be 
drop trigger schema_name.trigger_name